implementation/src/main/java/io/smallrye/config/RelocateConfigSourceInterceptor.java

```diff
@@ -0,0 +1,29 @@
+package io.smallrye.config;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.annotation.Priority;
+
+@Priority(300)
+public class RelocateConfigSourceInterceptor implements ConfigSourceInterceptor {
```

Comment:
Since these interceptors don't do anything unless they're manually configured, does it make sense to have them be added by default or discovered?  Maybe we need a defaulted `getPriority()` method on `ConfigSourceInterceptor`.

Reply:
I think it makes sense to have a method to set the priority as well like `ConfigSource`. 

BTW, these interceptors are not added by default or get discovered either. You always have to manually configure them.

Reply:
Hum, thinking a little better about this. The changes we made in https://github.com/smallrye/smallrye-config/pull/275 with the `io.smallrye.config.ConfigSourceInterceptorFactory` require us to have the interceptors priority before construction time to order the initialization oder. If we add a method to provide the priority, then we have no way to know the right priority before initialization.

Reply:
True, that presents a challenge.  Perhaps the priority would be set by the factory?

Reply:
Yes, I see no other way. We already do that indirectly with the method I added to retrieve the interceptor class.

Reply:
Ok, I've implemented this in https://github.com/smallrye/smallrye-config/pull/275, since I already had the support for the factory and the Interceptor with priority in the builder.

We need to merge https://github.com/smallrye/smallrye-config/pull/275 first so I can clean this up with the work done over there.